src/config/config_impl.go

```diff
@@ -305,6 +305,15 @@ func (this *rateLimitConfigImpl) GetLimit(
 		finalKey := entry.Key + "_" + entry.Value
 		logger.Debugf("looking up key: %s", finalKey)
 		nextDescriptor := descriptorsMap[finalKey]
+
+		if nextDescriptor == nil {
+			for descriptorKey := range descriptorsMap {
+				if descriptorKey[len(descriptorKey)-1:] == "*" && strings.HasPrefix(finalKey, strings.TrimSuffix(descriptorKey, "*")) {
+					nextDescriptor = descriptorsMap[descriptorKey]
+				}
+			}
+		}
```

Comment:
Instead of doing this on every lookup, can you pre-store whether value has the potential for a wildcard and only do this in that case? That will avoid the perf hit for all users.

Reply:
@mattklein123 Updated the PR. Please review and let me know if this is what you've expected to see.